loader/windows/icd_windows.c

```diff
@@ -97,6 +98,77 @@ void adapterFree(WinAdapter *pWinAdapter)
     pWinAdapter->szName = NULL;
 }
 
+#if defined(CL_ENABLE_LAYERS)
+typedef struct WinLayer
+{
+    char * szName;
+    DWORD priority;
+} WinLayer;
+
+static WinLayer* pWinLayerBegin;
+static WinLayer* pWinLayerEnd;
+static WinLayer* pWinLayerCapacity;
+
+static int compareLayer(const void *a, const void *b)
+{
+    return ((WinLayer *)a)->priority < ((WinLayer *)b)->priority ? -1 :
+           ((WinLayer *)a)->priority > ((WinLayer *)b)->priority ? 1 : 0;
+}
+
+static BOOL layerAdd(const char* szName, DWORD priority)
+{
+    BOOL result = TRUE;
+    if (pWinLayerEnd == pWinLayerCapacity)
+    {
+        size_t oldCapacity = pWinLayerCapacity - pWinLayerBegin;
+        size_t newCapacity = oldCapacity;
+        if (0 == newCapacity)
+        {
+            newCapacity = 1;
+        }
+        else if(newCapacity < UINT_MAX/2)
+        {
+            newCapacity *= 2;
+        }
+
+        WinLayer* pNewBegin = malloc(newCapacity * sizeof(*pWinLayerBegin));
+        if (!pNewBegin)
+            result = FALSE;
```

Comment:
If we print out-of-memory errors on Linux (in which case printing to the console is problematic on it's own), we should strive to be consistent and do the same on Windows.
```suggestion
        if (!pNewBegin)
        {
            KHR_ICD_TRACE("Failed allocate space for Layers array\n");
            result = FALSE;
        }
```